include/revng/PTML/Tag.h

```diff
@@ -0,0 +1,119 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <algorithm>
+#include <climits>
+
+#include "llvm/ADT/SmallString.h"
+#include "llvm/ADT/StringMap.h"
+
+#include "revng/ADT/Concepts.h"
+#include "revng/Support/Debug.h"
+#include "revng/Support/YAMLTraits.h"
+
+namespace ptml {
+
+class Tag {
+private:
+  std::string TheTag;
+  std::string Content;
+  llvm::StringMap<std::string> Attributes;
+
+public:
+  Tag() {}
+  explicit Tag(llvm::StringRef Tag) : TheTag(Tag.str()) {}
+  explicit Tag(llvm::StringRef Tag, llvm::StringRef Content) :
+    TheTag(Tag.str()), Content(Content.str()) {}
+
+  struct Scope {
+  private:
+    llvm::raw_ostream &OS;
+    const Tag &TheTag;
+
+  public:
+    Scope(llvm::raw_ostream &OS, const Tag &TheTag, bool Newline) :
+      OS(OS), TheTag(TheTag) {
+      OS << TheTag.open();
+      if (Newline)
+        OS << "\n";
+    }
+
+    ~Scope() { OS << TheTag.close(); }
+  };
```

Comment:
Nice. I'd take it out of `class Tag` and document with an example on how to use it.

Reply:
Done